descartes_core/include/descartes_core/robot_model.h

```diff
@@ -129,7 +129,12 @@ class RobotModel
    * @return                 [description]
    */
   virtual bool isValidMove(const std::vector<double> &from_joint_pose, const std::vector<double> &to_joint_pose,
-                           double dt) const = 0;
+                           double dt) const
+  {
+    return isValidMove(from_joint_pose.data(), to_joint_pose.data(), dt);
+  }
+
+  virtual bool isValidMove(const double* s, const double* f, double dt) const = 0;
```

Comment:
I agree. It's a public facing API that uses raw pointers and it's anything but foolproof. 

This change is completely a side-effect (that I had not considered when I started this), of my representation of data in memory.

I can not make reasonable comments about the performance of this alternative representation without measuring. I can say that:
 1. Having a vector per solution adds ~3 extra 64 bit ints per solution. These size-cost of vertices are far out-weighed by edges in finely discretized graphs.
 2. While a vector of vectors would be logically compact, it does not guarantee that the memory owned by the vectors themselves is contiguous. I was banking on the memory having good spatial locality to help with cache efficiency.

Finally, as I mentioned in the intro above, the cost of isValidMove checks represents a significant chunk of program time and does a lot of redundant math. I'd really like to do away with it all-together and just make the planning graph aware of joint velocity limits through the robot model base class.